### PR TITLE
Add eWasm externals.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `6cda14ef7c9cb1a9a4101935a82ff889f7911288`:
```
Error: Trailing whitespace found:
 test/cmdlineTests/standard_eWasm_requested/output.json:102: 
 test/cmdlineTests/standard_eWasm_requested/output.json:109: 

```
Please check that your changes are working as intended.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `738e89c0ca2ce263a6aa280cd107a200b15f4a32`:
```
Error: Trailing whitespace found:
 test/cmdlineTests/standard_eWasm_requested/output.json:105: 
 test/cmdlineTests/standard_eWasm_requested/output.json:112: 

```
Please check that your changes are working as intended.

File: libyul/backends/wasm/WasmDialect.cpp [hunk 1]
```diff
@@ -85,6 +87,69 @@ WasmDialect const& WasmDialect::instance()
...
+		{"call", {i64, i32ptr, i32ptr, i32ptr, i32}, {i32}},
+		{"callDataCopy", {i32ptr, i32, i32}, {}},
+		{"getCallDataSize", {}, {i32}},
+		{"callCode", {i64, i32ptr, i32ptr, i32ptr, i32}, {i32}},
+		{"callDelegate", {i64, i32ptr, i32ptr, i32}, {i32}},
+		{"callStatic", {i64, i32ptr, i32ptr, i32}, {i32}},
+		{"storageStore", {i32ptr, i32ptr}, {}},
+		{"storageLoad", {i32ptr, i32ptr}, {}},
+		{"getCaller", {i32ptr}, {}},
+		{"getCallValue", {i32ptr}, {}},
```
Comment (leonardoalt) [inline libyul/backends/wasm/WasmDialect.cpp:111]: Why doesn't this have a return?

**Review by leonardoalt — COMMENTED** [review]
Did not check in detail the wasm functions and output

File: libyul/backends/wasm/WasmDialect.cpp (hunk 1 above)
Comment (chriseth) [reply libyul/backends/wasm/WasmDialect.cpp:111]: If the return value is longer than 64 bits, it is stored in memory.

Comment (leonardoalt) [reply libyul/backends/wasm/WasmDialect.cpp:111]: oh ok

File: libyul/backends/wasm/WasmDialect.cpp [hunk 2]
```diff
@@ -85,6 +87,69 @@ WasmDialect const& WasmDialect::instance()
...
+	// These are not YulStrings because that would be too complicated with regards
+	// to the YulStringRepository reset.
+	static string const i64{"i64"};
+	static string const i32{"i32"};
+	static string const i32ptr{"i32"}; // Uses "i32" on purpose.
+	struct External { string name; vector<string> parameters; vector<string> returns; };
+	static vector<External> externals{
+		{"getAddress", {i32ptr}, {}},
+		{"getExternalBalance", {i32ptr, i32ptr}, {}},
+		{"getBlockHash", {i64, i32ptr}, {i32}},
```
Comment (leonardoalt) [inline libyul/backends/wasm/WasmDialect.cpp:101]: what's the output here?

File: libyul/backends/wasm/WasmDialect.cpp [hunk 3]
```diff
@@ -85,6 +87,69 @@ WasmDialect const& WasmDialect::instance()
...
+		{"getCaller", {i32ptr}, {}},
+		{"getCallValue", {i32ptr}, {}},
+		{"codeCopy", {i32ptr, i32, i32}, {}},
+		{"getCodeSize", {i32ptr}, {}},
+		{"getBlockCoinbase", {i32ptr}, {}},
+		{"create", {i32ptr, i32ptr, i32, i32ptr}, {i32}},
+		{"getBlockDifficulty", {i32ptr}, {}},
+		{"externalCodeCopy", {i32ptr, i32ptr, i32, i32}, {}},
+		{"getExternalCodeSize", {i32ptr}, {i32}},
+		{"getGasLeft", {}, {i64}},
```
Comment (leonardoalt) [inline libyul/backends/wasm/WasmDialect.cpp:119]: Is the return going to stay 64bits for gas left?

File: libyul/backends/wasm/WasmDialect.cpp (hunk 2 above)
Comment (chriseth) [reply libyul/backends/wasm/WasmDialect.cpp:101]: `getBlockhash(blockNr, outputPtr) -> success`

Comment (chriseth) [reply libyul/backends/wasm/WasmDialect.cpp:101]: Acutally output is 1 on failure and 0 on success.

File: libyul/backends/wasm/WasmDialect.cpp (hunk 3 above)
Comment (chriseth) [reply libyul/backends/wasm/WasmDialect.cpp:119]: That is how it is defined for now.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**